extensions-contrib/azure-extensions/src/test/java/org/apache/druid/data/input/azure/AzureInputSourceTest.java

```diff
@@ -0,0 +1,207 @@
+
+import com.google.common.collect.ImmutableList;
+import org.apache.druid.data.input.InputSplit;
+import org.apache.druid.data.input.impl.CloudObjectLocation;
+import org.apache.druid.data.input.impl.SplittableInputSource;
+import org.apache.druid.storage.azure.AzureCloudBlobHolderToCloudObjectLocationConverter;
+import org.apache.druid.storage.azure.AzureCloudBlobIterable;
+import org.apache.druid.storage.azure.AzureCloudBlobIterableFactory;
+import org.apache.druid.storage.azure.AzureStorage;
+import org.apache.druid.storage.azure.CloudBlobHolder;
+import org.easymock.EasyMock;
+import org.easymock.EasyMockSupport;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.net.URI;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Spliterators;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class AzureInputSourceTest extends EasyMockSupport
+{
+  private static final String CONTAINER_NAME = "container";
+  private static final String BLOB_NAME = "blob";
+  private static final URI PREFIX_URI;
+  private final List<URI> EMPTY_URIS = ImmutableList.of();
+  private final List<URI> EMPTY_PREFIXES = ImmutableList.of();
+  private final List<CloudObjectLocation> EMPTY_OBJECTS = ImmutableList.of();
+  private static final String CONTAINER = "CONTAINER";
+  private static final String BLOB_PATH = "BLOB_PATH";
+  private static final CloudObjectLocation CLOUD_OBJECT_LOCATION_1 = new CloudObjectLocation(CONTAINER, BLOB_PATH);
+
+  private AzureStorage storage;
+  private AzureEntityFactory entityFactory;
+  private AzureCloudBlobIterableFactory azureCloudBlobIterableFactory;
+  private AzureCloudBlobHolderToCloudObjectLocationConverter azureCloudBlobToLocationConverter;
+
+  private InputSplit<CloudObjectLocation> inputSplit;
+  private AzureEntity azureEntity1;
+  private CloudBlobHolder cloudBlobDruid1;
+  private AzureCloudBlobIterable azureCloudBlobIterable;
+
+  private AzureInputSource azureInputSource;
+
+  static {
+    try {
+      PREFIX_URI = new URI(AzureInputSource.SCHEME + "://" + CONTAINER_NAME + "/" + BLOB_NAME);
+    }
+    catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Before
+  public void setup()
+  {
+    storage = createMock(AzureStorage.class);
+    entityFactory = createMock(AzureEntityFactory.class);
+    inputSplit = createMock(InputSplit.class);
+    azureEntity1 = createMock(AzureEntity.class);
+    azureCloudBlobIterableFactory = createMock(AzureCloudBlobIterableFactory.class);
+    azureCloudBlobToLocationConverter = createMock(AzureCloudBlobHolderToCloudObjectLocationConverter.class);
+    cloudBlobDruid1 = createMock(CloudBlobHolder.class);
+    azureCloudBlobIterable = createMock(AzureCloudBlobIterable.class);
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void test_Constructor_emptyUrisEmptyPrefixesEmptyObjects_throwsIllegalArgumentException()
```

Comment:
nit: I prefere `test_init_...` but maybe `Constructor` should be `constructor` since it's the first word after `_`